File: 0274-h-index/0274-h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        arr = [0 for i in range(n+1)]
        for num in citations:
            if num >= n:
                arr[n] += 1
            else:
                arr[num] += 1
        ans = 0
        for i in reversed(range(n+1)):
            ans += arr[i]
            if ans >= i:
                return i
        return 0

File: 0274-h-index/0274-h-index.java
class Solution {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        int ans = citations[0] == 0 ? 0 : 1;
        int n = citations.length;
        for(int i=0; i<citations.length; i++) {
            if(citations[i] >= n-i) ans = Math.max(ans, n-i);
        }
        return ans;
    }
}